### PR TITLE
Google Cloud Run is the new Cloud Functions

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Changes Tempo's GCP serverless technology from Function to Run. Google Cloud Run supports generic container images and allows us to use Go 1.18 for Parquet support. 

Changes made:
- In drone upload a docker image instead of a zip file to gcs
- Remove cloud functions code/makefile instructions and replace with cloud run
- Updated e2e tests to use the new cloud run image
- Dropped a few places we were using custom code instead of 1.17 features (time.UnixMicro(), math.MaxInt)
- Changed Cloud Functions documentation to be Cloud Run focused
- Updated both Lambda and Cloud run to use Go 1.17 to match the rest of the codebase.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/trace_info.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (t *TraceInfo) makeThriftBatch(TraceIDHigh int64, TraceIDLow int64) *thrift
 			OperationName: fmt.Sprintf("vulture-%d", t.generateRandomInt(0, 100)),
 			References:    nil,
 			Flags:         0,
-			StartTime:     unixMicro(t.timestamp),
+			StartTime:     time.Now().UnixMicro(),
```
Comment (mdisibio) [inline pkg/util/trace_info.go:144]: Does this impact vulture's consistency checks?

Comment (joe-elliott) [reply pkg/util/trace_info.go:144]: This should be the exact same value:

https://github.com/grafana/tempo/pull/1483/files#diff-64607fbb74bbb31f1d923cc9c1b4916f90ba904eb09a21320dbf7f2cf2dc59fdL13-L16

Comment (joe-elliott) [reply pkg/util/trace_info.go:144]: I say this and I just noticed that tons of tests related to these timestamps are failing. Will dig deeper.

Comment (joe-elliott) [reply pkg/util/trace_info.go]: yup, i see what i did now. good catch!

File: cmd/tempo-serverless/cloud-run/go.mod [hunk 2]
```diff
@@ -0,0 +1,147 @@
+module github.com/grafana/tempo/cmd/tempo-serverless/cloud-run
+
+go 1.18
```
Comment (zalegrala) [inline cmd/tempo-serverless/cloud-run/go.mod:3]: Time to party.

File: cmd/tempo-serverless/cloud-run/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+#
+#  to exercise the function
+#    curl http://localhost:8080/?start=1000&end=1001&...
+#
+
+#
+# build the serverless container image
+#
+#  todo: FROM scratch saves ~5MB which could be meaningful in a serverless setting, but using scratch gave strange errors on query.
+FROM alpine:3.15 as certs
```
Comment (zalegrala) [inline cmd/tempo-serverless/cloud-run/Dockerfile:12]: I think 3.16 is out if you want to upgrade.

**Review by zalegrala — APPROVED** [review]
This looks good to me.  Thanks for putting it together.

File: cmd/tempo-serverless/cloud-run/go.mod (hunk 2 above)
Comment (joe-elliott) [reply cmd/tempo-serverless/cloud-run/go.mod:3]: I'm actually going to walk this back to 1.17 to match the rest of the codebase. We should move this forward together.

File: cmd/tempo-serverless/cloud-run/Dockerfile (hunk 3 above)
Comment (joe-elliott) [reply cmd/tempo-serverless/cloud-run/Dockerfile:12]: Not a bad idea, but all other images are on 3.15. I'm going to leave this on 3.15 as well. We should open a separate PR that moves all of them forward as one.

**Pull request merged**